### PR TITLE
feat(atomic): dynamically import dayjs locales

### PR DESCRIPTION
**Pull request by ThibodeauJF**
https://coveord.atlassian.net/browse/KIT-1610

### CODE REVIEW

File: packages/atomic/src/utils/day-js-locales.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import dayjs from 'dayjs';
+
+const locales: Record<string, () => Promise<unknown>> = {
```
Comment (ThibodeauJF) [inline packages/atomic/src/utils/day-js-locales.ts:3]: this automatically adds each files to the build & lazy loads the locales as needed

Comment (ThibodeauJF) [reply packages/atomic/src/utils/day-js-locales.ts:3]: The locales we support won't probably change in the future, but we could always...
- Put this file into .gitignore
- Add a pre-build step to parse the `locales.json` file or the `node_module/dayjs/locale` folder to dynamically add locales

File: packages/atomic/src/components/result-template-components/atomic-result-date/atomic-result-date.tsx [hunk 2]
```diff
@@ -17,6 +21,9 @@ export class AtomicResultDate {
 
   @State() public error!: Error;
 
+  @InitializeBindings()
+  public bindings!: Bindings;
```
Comment (liviarett) [inline packages/atomic/src/components/result-template-components/atomic-result-date/atomic-result-date.tsx:25]: why did you need to add this here? 🤔

**Review by btaillon — APPROVED** [review]

File: packages/atomic/src/components/result-template-components/result-template-decorators.tsx [hunk 3]
```diff
@@ -55,6 +55,14 @@ export function ResultContext(opts: {folded: boolean} = {folded: false}) {
...
     };
 
+    component.componentWillRender = function () {
+      if (this.error) {
+        return;
+      }
+
+      return componentWillRender && componentWillRender.call(this);
+    };
+
```
Comment (ThibodeauJF) [inline packages/atomic/src/components/result-template-components/result-template-decorators.tsx:65]: prevents console errors

File: packages/atomic/src/utils/day-js-locales.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  en: () => import('dayjs/locale/en'),
```
[the remark this answers is not in the document]
Comment (ThibodeauJF) [reply packages/atomic/src/utils/day-js-locales.ts:4]: yes https://github.com/coveo/ui-kit/pull/1993/files#r874208015

[the remark this answers is not in the document]
Comment (olamothe) [reply packages/atomic/src/utils/day-js-locales.ts:4]: To fix the problem, I just removed the dynamic theme imports from atomic-react 😂
